lib/samples/show_grid/show_grid_sample.dart

```diff
@@ -0,0 +1,430 @@
+            onLabelColorChanged: _onLabelColorChanged,
+            onLabelPositionChanged: _onLabelPositionChanged,
+            onLabelFormatChanged: _onLabelFormatChanged,
+            onLabelVisibilityChanged: _onLabelVisibilityChanged,
+          ),
+          actions: [
+            TextButton(
+              child: const Text('Close'),
+              onPressed: () => Navigator.of(context).pop(),
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+///
+/// a widget that displays grid options.
+///
+class GridOptions extends StatefulWidget {
+  final Function(Grid) onGridChanged;
+  final Function(Color) onGridColorChanged;
+  final Function(Color) onLabelColorChanged;
+  final Function(GridLabelPosition) onLabelPositionChanged;
+  final Function(LatitudeLongitudeGridLabelFormat) onLabelFormatChanged;
+  final Function(bool) onLabelVisibilityChanged;
+
+  const GridOptions({
+    super.key,
+    required this.onGridChanged,
+    required this.onGridColorChanged,
+    required this.onLabelColorChanged,
+    required this.onLabelPositionChanged,
+    required this.onLabelFormatChanged,
+    required this.onLabelVisibilityChanged,
+  });
+
+  @override
+  State<GridOptions> createState() => _GridOptionsState();
+}
+
+class _GridOptionsState extends State<GridOptions> with SampleStateSupport {
+  final grids = ['Latitude & Longitude', 'MGRS', 'UTM', 'USNG'];
+  final colors = ['Red', 'Blue', 'Green', 'Yellow'];
+  final labelPositions = [
+    'AllSides',
+    'BottomLeft',
+    'BottomRight',
+    'Center',
+    'Geographic',
+    'TopLeft',
+    'TopRight'
+  ];
+  final latitudeLongitudeGridLabelFormats = [
+    'Decimal Degrees',
+    'Degrees Minutes Seconds'
+  ];
+
+  // stateful variables
+  String? grid;
+  String? gridColor;
+  String? labelColor;
+  String? labelPosition;
+  String? labelFormat;
+  var isLabelFormatVisible = true;
+  var labelVisible = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      child: ListBody(
+        children: [
+          _buildGridDropdown(),
+          isLabelFormatVisible
+              ? _buildLatLongLabelFormatDropdown()
+              : Container(),
+          _buildGridColorDropdown(),
+          _buildLabelColorDropdown(),
+          _buildLabelPositionDropdown(),
+          _buildLabelVisibilityCheckbox(),
+        ],
+      ),
+    );
+  }
+
+  // Create a DropdownButtonFormField widget.
+  static DropdownButtonFormField _createDropdownButtonFormField({
+    required String value,
+    required String labelText,
+    required List<String> items,
+    required Function onChanged,
+  }) {
+    return DropdownButtonFormField<String>(
+      value: value,
+      onChanged: (newValue) {
+        onChanged(newValue!);
+      },
+      decoration: InputDecoration(
+        labelText: labelText,
+      ),
+      items: items.map<DropdownMenuItem<String>>((value) {
+        return DropdownMenuItem<String>(
```

Comment:
I think these two lines can be simplified to:
```dart
  items: items.map((value) {
    return DropdownMenuItem(
```
I think the compiler will still be able to infer the proper type, but I'm not 100% sure.